File: Problems/No.62_Unique_Paths/unique_paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """

        # parameter
        AREA = [[1]*n for i in range(m)]

        # start computing
        for i in range(1, m):
            for j in range(1, n):
                AREA[i][j] = AREA[i-1][j] + AREA[i][j-1]
        
        return AREA[-1][-1]

File: Problems/No.62_Unique_Paths/unique_paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        
        // parameters
        std::vector<std::vector<int>> AREA(m, std::vector<int> (n, 1));

        // start computing
        for (int i=1; i<m; i++)
            for (int j=1; j<n; j++)
                AREA[i][j] = AREA[i-1][j] + AREA[i][j-1];

        return AREA[m-1][n-1];

    }
};
